server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java

```diff
@@ -103,20 +103,22 @@ CassandraSessionConfiguration getCassandraSessionConfiguration(PropertiesProvide
 
     @Provides
     @Singleton
-    FetchBatchSizes getFetchBatchSizesConfiguration(PropertiesProvider propertiesProvider) {
+    BatchSizes getBatchSizesConfiguration(PropertiesProvider propertiesProvider) {
         try {
             PropertiesConfiguration configuration = propertiesProvider.getConfiguration(BATCHSIZES_FILE_NAME);
-            FetchBatchSizes fetchBatchSizes = FetchBatchSizes.builder()
-                    .metadata(configuration.getInt("metadata", FetchBatchSizes.DEFAULT_FETCH_BATCH_SIZE))
-                    .headers(configuration.getInt("headers", FetchBatchSizes.DEFAULT_FETCH_BATCH_SIZE))
-                    .body(configuration.getInt("body", FetchBatchSizes.DEFAULT_FETCH_BATCH_SIZE))
-                    .full(configuration.getInt("full", FetchBatchSizes.DEFAULT_FETCH_BATCH_SIZE))
+            BatchSizes batchSizes = BatchSizes.builder()
```

Comment:
I think the solution I gave here: https://github.com/linagora/james-project/pull/803#discussion_r117748391is short to put in place and not that ugly. I don't see why we should wait for Java8 by the way this code is in the Cassandra module there is already Java8